include/OpenSpaceToolkit/Astrodynamics/Flight/System/SatelliteDynamics.hpp

```diff
@@ -0,0 +1,161 @@
+
+#include <OpenSpaceToolkit/Physics/Environment/Objects/CelestialBodies/Moon.hpp>
+#include <OpenSpaceToolkit/Physics/Environment/Objects/CelestialBodies/Sun.hpp>
+#include <OpenSpaceToolkit/Physics/Environment/Objects/CelestialBodies/Earth.hpp>
+#include <OpenSpaceToolkit/Physics/Environment.hpp>
+#include <OpenSpaceToolkit/Physics/Environment/Objects/Celestial.hpp>
+#include <OpenSpaceToolkit/Physics/Data/Vector.hpp>
+#include <OpenSpaceToolkit/Physics/Units/Derived.hpp>
+#include <OpenSpaceToolkit/Physics/Units/Length.hpp>
+#include <OpenSpaceToolkit/Physics/Time/Instant.hpp>
+#include <OpenSpaceToolkit/Physics/Time/Duration.hpp>
+
+#include <OpenSpaceToolkit/Core/Containers/Array.hpp>
+#include <OpenSpaceToolkit/Core/Types/String.hpp>
+#include <OpenSpaceToolkit/Core/Types/Real.hpp>
+#include <OpenSpaceToolkit/Core/Types/Integer.hpp>
+#include <OpenSpaceToolkit/Core/Types/Shared.hpp>
+
+////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////
+
+namespace ostk
+{
+namespace astro
+{
+namespace flight
+{
+namespace system
+{
+
+////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////
+
+using ostk::core::types::Integer ;
+using ostk::core::types::Real ;
+using ostk::core::types::String ;
+using ostk::core::types::Shared ;
+using ostk::core::ctnr::Array ;
+
+using ostk::math::obj::Vector3d ;
+
+using ostk::physics::Unit ;
+using ostk::physics::time::Instant ;
+using ostk::physics::time::Duration ;
+using ostk::physics::units::Length ;
+using ostk::physics::units::Mass ;
+using ostk::physics::units::Derived ;
+using ostk::physics::coord::Position ;
+using ostk::physics::coord::Velocity ;
+using ostk::physics::coord::Frame ;
+using ostk::physics::env::obj::Celestial ;
+using ostk::physics::env::Object ;
+using ostk::physics::Environment ;
+using ostk::physics::env::obj::celest::Earth ;
+using ostk::physics::env::obj::celest::Sun ;
+using ostk::physics::env::obj::celest::Moon ;
+using ostk::physics::data::Vector ;
+
+using ostk::astro::flight::SatelliteSystem ;
+using ostk::astro::trajectory::State ;
+
+////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////
+
+/// @brief                      Defines a satellite in orbit subject to forces of varying fidelity. Represents a system of differential equationals that can be solved by calling the NumericalSolver class.
+
+class SatelliteDynamics
+{
+
+    public:
+
+        enum class StateVectorType // Enum class values to represent dimension of state vector
+        {
+            PositionVelocity = 6,
+            PositionVelocitywithDragCoefficient = 7
+        } ;
+
+        typedef std::vector<double> StateVector ; // Container used to hold the state vector
```

Comment:
Not sure about this, boost recommends the use of std::vector so that's why I went with it. This could be included in a refactor at a later date if we find that an Eigen vector3d performs better than an std::vector